sumpy/expansion/local.py

```diff
@@ -682,6 +668,42 @@ def translate_from(self, src_expansion, src_coeff_exprs, src_rscale,
         logger.info("building translation operator: done")
         return result
 
+    def loopy_translate_from(self, src_expansion):
+        from sumpy.expansion.multipole import VolumeTaylorMultipoleExpansionBase
+
+        if isinstance(src_expansion, VolumeTaylorMultipoleExpansionBase):
+            ncoeff_src = self.m2l_preprocess_multipole_nexprs(src_expansion)
+            ncoeff_tgt = self.m2l_postprocess_local_nexprs(src_expansion)
+
+            icoeff_src = pymbolic.var("icoeff_src")
+            icoeff_tgt = pymbolic.var("icoeff_tgt")
+            domains = [f"{{[icoeff_tgt]: 0<=icoeff_tgt<{ncoeff_tgt} }}"]
+
+            src_coeffs = pymbolic.var("src_coeffs")
+            m2l_translation_classes_dependent_data = pymbolic.var("data")
+
+            if self.use_preprocessing_for_m2l:
+                expr = src_coeffs[icoeff_tgt] \
+                        * m2l_translation_classes_dependent_data[icoeff_tgt]
+            else:
+                toeplitz_first_row = src_coeffs[icoeff_src-icoeff_tgt]
+                vector = m2l_translation_classes_dependent_data[icoeff_src]
+                expr = toeplitz_first_row * vector
+                domains += [
+                        f"{{[icoeff_src]: icoeff_tgt<=icoeff_src<{ncoeff_src} }}"]
+
+            insns = f"""
+            coeff[icoeff_tgt] = coeff[icoeff_tgt] + {expr}
+            """
```

Comment:
Write as `lp.Assignment`.

Reply:
Done in 1e3f1e5c494121f704ce34ffde21c395704bd424